tasks/src/main/java/error_exceptions/service/StudentsContainer.java

```diff
@@ -0,0 +1,134 @@
+package error_exceptions.service;
+
+import error_exceptions.exceptions.GroupsInFacultyException;
+import error_exceptions.exceptions.StudentInGroupException;
+import error_exceptions.exceptions.StudentSubjectsException;
+import error_exceptions.models.Faculty;
+import error_exceptions.models.Group;
+import error_exceptions.models.Student;
+
+import error_exceptions.exceptions.FacultyInUniversityException;
+import java.util.List;
+
+public class StudentsContainer {
+
+    public static final Student studentMillsSteven = new Student("Mills", "Steven");
+    public static final Student studentMillsVincent = new Student("Mills", "Vincent");
+    public static final Student studentAndrewsBetty = new Student("Andrews", "Betty");
+    public static final Student studentHarrisJulian = new Student("Harris", "Julian");
+    public static final Student studentBensonMary = new Student("Benson", "Mary");
+    public static final Student studentWadeAnthony = new Student("Wade", "Anthony");
+
+    private static double averageMark = 0.0;
+    private static int countSubjects = 0;
+    private static double sumAllMarks = 0.0;
+
+    private static double averageMark(Student student) {
+        sumAllMarks += student.getMark();
+        if (student.getSubject() != null) {
+            countSubjects++;
+        }
+        return sumAllMarks / countSubjects;
+    }
+
+    public static double averageMarkOfAllSubjects(List<Student> students, String lastName,
+        String firstName) {
+        for (Student student : students) {
+            if (student.getLastName().equals(lastName) && student.getFirstName()
+                .equals(firstName)) {
+                averageMark = averageMark(student);
+            }
+        }
+        if (averageMark == 0.0) {
+            throw new StudentSubjectsException("Student " + lastName + " " + firstName
+                + " without marks or not study in this university");
+        }
+        return averageMark;
+    }
+
+    private static boolean studentInGroup(List<Faculty> faculties, Group.GroupName group) {
+        boolean hasStudent = false;
+        for (Faculty faculty : faculties) {
+            String groupNames = faculty.getGroupName().name();
+            if (groupNames.contains(group.name())) {
+                hasStudent = true;
```

Comment:
Are you sure about the logic of the method? It doesn't check any student. Why if a faculty group name contains another group name you decide that it has a student?
You just need to check if a group contains a student